crates/modelardb_storage/src/delta_lake.rs

```diff
@@ -0,0 +1,495 @@
+            .with_storage_options(self.storage_options.clone())
+            .with_table_name(table_name)
+            .with_location(location)
+            .with_columns(columns)
+            .with_partition_columns(partition_columns)
+            .with_save_mode(save_mode)
+            .await
+            .map_err(|error| error.into())
+    }
+
+    /// Drop the metadata Delta Lake table with `table_name` from the Delta Lake by deleting every
+    /// file related to the table. The table folder cannot be deleted directly since folders do not
+    /// exist in object stores and therefore cannot be operated upon. If the table was dropped
+    /// successfully, the paths to the deleted files are returned, otherwise a
+    /// [`ModelarDbStorageError`] is returned.
+    pub async fn drop_metadata_delta_lake_table(&self, table_name: &str) -> Result<Vec<Path>> {
+        let table_path = format!("{METADATA_FOLDER}/{table_name}");
+        self.delete_table_files(&table_path).await
+    }
+
+    /// Drop the Delta Lake table with `table_name` from the Delta Lake by deleting every file
+    /// related to the table. The table folder cannot be deleted directly since folders do not exist
+    /// in object stores and therefore cannot be operated upon. If the table was dropped
+    /// successfully, the paths to the deleted files are returned, otherwise a
+    /// [`ModelarDbStorageError`] is returned.
+    pub async fn drop_delta_lake_table(&self, table_name: &str) -> Result<Vec<Path>> {
+        let table_path = format!("{COMPRESSED_DATA_FOLDER}/{table_name}");
+        self.delete_table_files(&table_path).await
+    }
+
+    /// Delete all files in the folder at `table_path` using bulk operations if available. If the
+    /// files were deleted successfully, the paths to the deleted files are returned.
+    async fn delete_table_files(&self, table_path: &str) -> Result<Vec<Path>> {
+        let file_locations = self
+            .object_store
+            .list(Some(&Path::from(table_path)))
+            .map_ok(|object_meta| object_meta.location)
+            .boxed();
+
+        let deleted_paths = self
+            .object_store
+            .delete_stream(file_locations)
+            .try_collect::<Vec<Path>>()
+            .await?;
+
+        Ok(deleted_paths)
+    }
+
+    /// Truncate the Delta Lake table with `table_name` by deleting all rows in the table. If the
+    /// rows could not be deleted, a [`ModelarDbStorageError`] is returned.
+    pub async fn truncate_delta_lake_table(&self, table_name: &str) -> Result<()> {
+        let delta_table_ops = self.delta_ops(table_name).await?;
+        delta_table_ops.delete().await?;
+
+        Ok(())
+    }
+
+    /// Write `rows` to a metadata Delta Lake table with `table_name`. Returns an updated
+    /// [`DeltaTable`] version if the file was written successfully, otherwise returns
+    /// [`ModelarDbStorageError`].
+    pub async fn write_rows_to_metadata_delta_table(
```

Comment:
I have kept it as it was since I would expect the three methods to take the same arguments if they had the `write_TABLE_TYPE` name. Keeping what is written in the method name makes it very explicit that there is a difference in the interface.